### PR TITLE
Bugfix: initializing all tensors and parameters of the `ConditionalRandomField` model on the proper device

### LINKED ISSUE #2884: Why CRF lead a high  cost on CPU?
I train two mode to do NER task:
1. 1 layer Bi-LSTM (crf_tagger) 
2. 1 layer Bi-LSTM+CRF (sample_tagger) 

I compare their cost on both CPU and  GPU  in training process. 
Here are the result:

| Model       | CPU(% )  | GPU(%)     |
| ---------- | :-----------:  | :-----------: |
|Bi-LSTM+CRF(AllenNLP)  | 1124 | 1 |
| Bi-LSTM(AllenNLP) |57 | 6 |
| Bi-LSTM+CRF(Keras) |235 | 13 |

It's a Linux server with 16 logical core (4 physical) single Tesla V100.
I use `top->%cpu` to get CPU cost and `nvidia-smi ->Volatile GPU Util` for GPU.
Intuitively, seems that `viterbi_decode ` lead this phenomenon , I read the source code but can't find how to solve it.
Can somebody help me ?

### PR DESCRIPTION
**Pull request by oroszgy**
Bugfix: `crf_tagger` was doing most of the computations on the CPU, as the `ConditionalRandomField` model's parameters and some necessary tensors were initialized on the wrong device.

Fixes #2884 .

Changes proposed in this pull request:

- `ConditionalRandomField`'s `transitions` and `tag_sequence` tensors are initialized on the proper device

## Before submitting

- [x] I've read and followed all steps in the [Making a pull request](https://github.com/allenai/allennlp/blob/main/CONTRIBUTING.md#making-a-pull-request)
    section of the `CONTRIBUTING` docs.
- [x] I've updated or added any relevant docstrings following the syntax described in the
    [Writing docstrings](https://github.com/allenai/allennlp/blob/main/CONTRIBUTING.md#writing-docstrings) section of the `CONTRIBUTING` docs.
- [ ] If this PR fixes a bug, I've added a test that will fail without my fix.
- [ ] If this PR adds a new feature, I've added tests that sufficiently cover my new functionality.

## After submitting

- [ ] All GitHub Actions jobs for my pull request have passed.
- [ ] **`codecov/patch`** reports high test coverage (at least 90%).
    You can find this under the "Actions" tab of the pull request once the other checks have finished.

### CODE REVIEW

File: allennlp/modules/conditional_random_field.py [hunk 1]
```diff
@@ -364,7 +364,7 @@ def viterbi_tags(
         # Augment transitions matrix with start and end transitions
         start_tag = num_tags
         end_tag = num_tags + 1
-        transitions = torch.Tensor(num_tags + 2, num_tags + 2).fill_(-10000.0)
+        transitions = torch.empty(num_tags + 2, num_tags + 2, device=logits.device).fill_(-10000.0)
```
Comment (dirkgr) [inline allennlp/modules/conditional_random_field.py:367]: This should use `torch.full()`, so we can create the tensor in one go.

Comment (dirkgr) [reply allennlp/modules/conditional_random_field.py:367]: It will also be faster!

**Review by dirkgr — CHANGES_REQUESTED** [review]
Just one little change, and this is ready to go!

**Comment by dirkgr** [comment]
Ah, it also needs an entry in the changelog. So two little changes I guess.

File: allennlp/modules/conditional_random_field.py (hunk 1 above)
Comment (oroszgy) [reply allennlp/modules/conditional_random_field.py]: Fixed: https://github.com/allenai/allennlp/pull/5335/commits/84d4e20282ef05a16762cf322683e5b3512db8ab

**Comment by oroszgy** [comment]
> Ah, it also needs an entry in the changelog. So two little changes I guess.

Done: https://github.com/allenai/allennlp/pull/5335/commits/7ffd9aedb43df1316df1b2365fcd84dbfacaa7cb

**Review by dirkgr — APPROVED** [review]

**Comment by dirkgr** [comment]
Thank you!

**Pull request merged**